src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/visualmode/VisualMode.java

```diff
@@ -292,25 +317,42 @@ public void execute(JsObject obj)
                {
                   PanmirrorCode markdown = Js.uncheckedCast(obj);
                   rv.arrive(() -> {
-                     if (markdown == null) {
+                     if (markdown == null)
+                     {
                         // note that ready.execute() is never called in the error case
                         return;
                      }
                      
+                     // we are about to mutate the document, so create a single
```

Comment:
You may have some pretty funky merge conflicts here as I took part of this function and wrapped it in a handler that does some prompting the first time visual mode is activated and/or when we may plow over column or sentence based line wrapping. Here's the gist of my change: https://github.com/rstudio/rstudio/commit/0c4edf4ddebd7ef7d840443ae3538ca6c3f8cd90#diff-06b7d8ba9d7ace26762c2194678a2fd8

Reply:
Thanks, will attempt to resolve and get your review when complete!